clients/src/test/java/org/apache/kafka/clients/MetadataTest.java

```diff
@@ -1323,7 +1323,7 @@ public void testConcurrentUpdateAndFetchForSnapshotAndCluster() throws Interrupt
                 } else { // Thread to read metadata snapshot, once its updated
                     try {
                         if (!atleastMetadataUpdatedOnceLatch.await(5, TimeUnit.MINUTES)) {
```

Comment:
How about using `assertDoesNotThrow`? For example:
```java
                    assertTrue(assertDoesNotThrow(() -> atleastMetadataUpdatedOnceLatch.await(5, TimeUnit.MINUTES)),
                            "Test had to wait more than 5 minutes, something went wrong.");
```

Reply:
A bit unusual to chain assert calls but I'll accept it